tests/integration-tests/conftest.py

```diff
@@ -202,22 +206,49 @@ def pcluster_config_reader(test_datadir, vpc_stacks, region, request):
     The current renderer already replaces placeholders for current keys:
         {{ region }}, {{ os }}, {{ instance }}, {{ scheduler}}, {{ key_name }},
         {{ vpc_id }}, {{ public_subnet_id }}, {{ private_subnet_id }}
+    The current renderer injects options for custom templates and packages in case these
+    are passed to the cli and not present already in the cluster config.
 
     :return: a _config_renderer(**kwargs) function which gets as input a dictionary of values to replace in the template
     """
     config_file = "pcluster.config.ini"
 
     def _config_renderer(**kwargs):
+        config_file_path = test_datadir / config_file
+        _add_custom_packages_configs(config_file_path, request)
         default_values = _get_default_template_values(vpc_stacks, region, request)
         file_loader = FileSystemLoader(str(test_datadir))
         env = Environment(loader=file_loader)
         rendered_template = env.get_template(config_file).render(**{**kwargs, **default_values})
-        (test_datadir / config_file).write_text(rendered_template)
-        return test_datadir / config_file
+        config_file_path.write_text(rendered_template)
+        return config_file_path
 
     return _config_renderer
 
 
+def _add_custom_packages_configs(cluster_config, request):
+    config = configparser.ConfigParser()
+    config.read(cluster_config)
+    cluster_template = "cluster {0}".format(config.get("global", "cluster_template", fallback="default"))
+
+    for custom_option in ["template_url", "custom_awsbatch_template_url", "custom_chef_cookbook"]:
+        if request.config.getoption(custom_option) and custom_option not in config[cluster_template]:
```

Comment:
It seems the value configured in the file takes precedence over the option passed by command line. 
Is that what you wanted? If yes could you add a comment in the readme?

Reply:
Yes that is the intended behaviour. I left a comment where I define the pcluster_config_reader: The current renderer injects options for custom templates and packages in case these are passed to the cli and not present already in the cluster config.

Will clarify in the README as well